components/Sidebar.go

```diff
@@ -0,0 +1,281 @@
+package components
+
+import (
+	"github.com/gdamore/tcell/v2"
+	"github.com/rivo/tview"
+
+	"github.com/jorgerojas26/lazysql/app"
+	"github.com/jorgerojas26/lazysql/commands"
+	"github.com/jorgerojas26/lazysql/models"
+)
+
+type SidebarState struct {
+	currentFieldIndex int
+}
+
+type Sidebar struct {
+	*tview.Frame
+	Flex         *tview.Flex
+	state        *SidebarState
+	FieldHeights []int
+	subscribers  []chan models.StateChange
+}
+
+func NewSidebar() *Sidebar {
+	flex := tview.NewFlex().SetDirection(tview.FlexColumnCSS)
+	frame := tview.NewFrame(flex)
+	frame.SetBackgroundColor(app.Styles.PrimitiveBackgroundColor)
+	frame.SetBorder(true)
+	frame.SetBorders(0, 0, 0, 0, 0, 0)
+
+	sidebarState := &SidebarState{
+		currentFieldIndex: 0,
+	}
+
+	newSidebar := &Sidebar{
+		Frame:       frame,
+		Flex:        flex,
+		state:       sidebarState,
+		subscribers: []chan models.StateChange{},
+	}
+
+	newSidebar.SetInputCapture(newSidebar.inputCapture)
+
+	newSidebar.SetBlurFunc(func() {
+		newSidebar.SetCurrentFieldIndex(0)
+	})
+
+	return newSidebar
+}
+
+func (sidebar *Sidebar) AddField(title, text string, fieldWidth int) {
+	field := tview.NewTextArea()
+	field.SetWrap(true)
+	field.SetDisabled(true)
+
+	field.SetBorder(true)
+	field.SetTitle(title)
+	field.SetTitleAlign(tview.AlignLeft)
+	field.SetTitleColor(app.Styles.PrimaryTextColor)
+	field.SetText(text, true)
+	field.SetTextStyle(tcell.StyleDefault.Background(app.Styles.PrimitiveBackgroundColor).Foreground(tview.Styles.SecondaryTextColor))
+
+	textLength := len(field.GetText())
+
+	itemFixedSize := 3
+
+	if textLength >= fieldWidth*3 {
+		itemFixedSize = 5
+	} else if textLength >= fieldWidth {
+		itemFixedSize = 4
+	} else {
+		field.SetWrap(false)
+	}
+
+	field.SetFocusFunc(func() {
+		_, y, _, h := field.GetRect()
+		_, _, _, mph := sidebar.GetRect()
+
+		if y >= mph {
+			hidingFieldIndex := 0
+			fieldCount := sidebar.Flex.GetItemCount()
+
+			for i := 0; i < fieldCount; i++ {
```

Comment:
This is a candidate for `for range` loop, I think

```suggestion
			for i := range fieldCount {
```

Reply:
I will need to upgrade go version in go.mod from 1.20 to 1.22. I don't know how much of it is a problem.

Reply:
You are providing a tool, not a lib.

So upgrading to 1.23 directly should be fine.

You will have to upgrade a lot of thing such as golangci-lint in the CI

While it's a thing to do, there is no need to do it in this PR, just to use `for range`

So, yes, you could upgrade, but not necessarily now